js/reference.js

```diff
@@ -63,12 +68,27 @@ var ERMrest = (function(module) {
             // validate the model references in the `uri` parameter
             // this method needs to internally construct a reference object that
             // represents the `uri` parameter
-            notimplemented();
+
+            // build reference
+            var reference = new Reference(location.href, location);
+            var server = this.getServer(reference._serviceUrl);
+            server.catalogs.get(reference._catalogId).then(function success(catalog) {
+                // Should I make a setter here?
+                reference.catalog = catalog;
```

Comment:
same for `reference._schema`, `..._table`, and so on.

general clarification... my skeleton api should not be changed... if you see something missing that I maybe got it wrong but we should discuss it.